src/Tool.TTFMetrics/MainViewModel.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Media3D;
+using Microsoft.Win32;
+using Microsoft.Xaml.Behaviors.Core;
+
+namespace Tool.TTFMetrics;
+
+public sealed class MainViewModel : INotifyPropertyChanged
+{
+    private string _content = string.Empty;
+    private string _filePath = string.Empty;
+
+    public MainViewModel()
+    {
+        SelectFileCommand = new ActionCommand(SelectFile);
+    }
+
+    public string Content
+    {
+        get => _content;
+        private set => SetField(ref _content, value);
+    }
+
+    public string FilePath
+    {
+        get => _filePath;
+        private set
+        {
+            if (SetField(ref _filePath, value))
+                Content = BuildContent(_filePath);
+        }
+    }
+
+    public ICommand SelectFileCommand { get; }
+
+    public event PropertyChangedEventHandler? PropertyChanged;
+
+    private static string BuildContent(string filePath)
+    {
+        if (string.IsNullOrWhiteSpace(filePath) || !File.Exists(filePath))
+            return string.Empty;
+
+        try
+        {
+            var uri = new Uri(filePath);
+
+            var gtf = new GlyphTypeface(uri);
+
+            var sb = new StringBuilder();
+
+            var xHeight = gtf.XHeight;
+
+            sb.AppendLine($@"xHeight=""{xHeight}""");
+
+            foreach (var (code, index) in gtf.CharacterToGlyphMap)
+            {
+                if (code > 255)
+                    continue;
+
+                var bbh = gtf.Height;
+                var bsb = gtf.BottomSideBearings[index];
+                var tsb = gtf.TopSideBearings[index];
+                var fullHeight = bbh - bsb - tsb;
+
+                var depth = gtf.DistancesFromHorizontalBaselineToBlackBoxBottom[index];
```

Comment:
I do believe that it should be `DistancesFromHorizontalBaselineToBlackBoxBottom ± bsb` but the rendering in the example is then broken, the selection box are not around the gliphs

Reply:
Should they? Current render looks good to me.